app/src/org/commcare/android/util/MediaUtil.java

```diff
@@ -1,82 +1,260 @@
+                    // scaling down
+                    return getBitmapScaledToContainer(imageFile, boundingHeight, boundingWidth);
+                } else {
+                    // scaling up
+                    return attemptBoundedScaleUp(imageFile, calculatedHeight, calculatedWidth,
+                            containerHeight, containerWidth);
+                }
+            }
+        } catch (InvalidReferenceException e) {
+            Log.e("ImageInflater", "image invalid reference exception for " + e.getReferenceString());
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    private static double computeInflationScaleFactor(Context context, int targetDensity) {
+        DisplayMetrics metrics = context.getResources().getDisplayMetrics();
+        final int SCREEN_DENSITY = metrics.densityDpi;
+
+        double actualNativeScaleFactor = metrics.density;
+        // The formula below is what Android *usually* uses to compute the value of metrics.density
+        // for a device. If this is in fact the value being used, we are not interested in it.
+        // However, if the actual value differs at all from the standard calculation, it means
+        // Android is taking other factors into consideration (such as straight up screen size),
+        // and we want to incorporate that proportionally into our own version of the scale factor
+        double standardNativeScaleFactor = (double)SCREEN_DENSITY / DisplayMetrics.DENSITY_DEFAULT;
+        double proportionalAdjustmentFactor = 1;
+        if (actualNativeScaleFactor > standardNativeScaleFactor) {
+            proportionalAdjustmentFactor = 1 +
+                    ((actualNativeScaleFactor - standardNativeScaleFactor) / standardNativeScaleFactor);
+        } else if (actualNativeScaleFactor < standardNativeScaleFactor) {
+            proportionalAdjustmentFactor = actualNativeScaleFactor / standardNativeScaleFactor;
+        }
+        Log.i("10/15", "proportional adjustment factor: " + proportionalAdjustmentFactor);
+
+        // Get our custom scale factor, based on this device's density and what the image's target
+        // density was
+        Log.i("10/15", "Target dpi: " + targetDensity);
+        Log.i("10/15", "This screen's dpi: " + SCREEN_DENSITY);
+        double customDpiScaleFactor = (double)SCREEN_DENSITY / targetDensity;
+        Log.i("10/15", "dpi scale factor: " + customDpiScaleFactor);
+
+        Log.i("10/15", "FINAL scale factor: " + (customDpiScaleFactor * proportionalAdjustmentFactor));
+        return customDpiScaleFactor * proportionalAdjustmentFactor;
     }
 
+
     /**
-     * Pass in a string representing either a GeoPont or an address and get back a valid
+     * Pass in a string representing either a GeoPoint or an address and get back a valid
      * GeoURI that can be passed as an intent argument 
      */
     public static String getGeoIntentURI(String rawInput){

```

Comment:
This feels like the wrong place for this method, but didn't have ideas for a better util class to move it to. If you have any thoughts, let me know.